core/src/components/button/button.tsx

```diff
@@ -120,6 +120,12 @@ export class Button implements ComponentInterface, AnchorInterface, ButtonInterf
    */
   @Prop() type: 'submit' | 'reset' | 'button' = 'button';
 
+  /**
+   * if `form` is defined, on button click the specified form will be submitted.
+   * Useful when `ion-button` is outside of the form.
+   */
+  @Prop() form: HTMLFormElement | undefined;
```

Comment:
The `form` property should accept a string, not a form element reference. This is similar to how the native `button` works.
```suggestion
  /**
   * The ID of the `<form>` element that should be
   * submitted on button click.
   */
  @Prop() form: string | undefined;
```